runtime/master/src/main/java/org/apache/nemo/runtime/master/scheduler/TaskDispatcher.java

```diff
@@ -205,7 +205,7 @@ void signal() {
     void await() {
       lock.lock();
       try {
-        if (!hasDelayedSignal) {
+        while (!hasDelayedSignal) {
```

Comment:
How about to add a simple comment that this while loop is to deal with spurious wakeup?